test/http_downloads_test.rb

```diff
@@ -7,8 +7,14 @@ def command(src, dst)
 end
 
 class HttpDownloadsTest < Test::Unit::TestCase
+  def test_should_return_an_instance
+    assert_instance_of Proxy::HttpDownload, Proxy::HttpDownloads.start_download('source', 'destination')
+  ensure
+    File.delete('destination')
```

Comment:
Most definitely. Joining a thread isn't great, as it will slow down tests. It would be worth it to break up HttpDownloads instantiation and starting, but is probably out of scope for this PR. Stubbing of HttpDownload constructor should work (ugh).